app/controllers/discover.js

```diff
@@ -0,0 +1,170 @@
+        this._super(...arguments);
+        this.set('facetFilters', Ember.Object.create());
+        this.loadPage();
+    },
+
+    loadPage() {
+        let queryBody = JSON.stringify(this.getQueryBody());
+        this.set('loading', true);
+        return Ember.$.ajax({
+            'url': this.get('searchUrl'),
+            'crossDomain': true,
+            'type': 'POST',
+            'contentType': 'application/json',
+            'data': queryBody
+        }).then((json) => {
+            this.set('numberOfResults', json.hits.total);
+//            this.set('took', moment.duration(json.took).asSeconds());
+            let results = json.hits.hits.map((hit) => {
+                // HACK
+                let source = hit._source;
+                source.id = hit._id;
+                source.type = 'elastic-search-result';
+                source.workType = source['@type'];
+                source.contributors = source.contributors.map(contributor => {
+                    return {
+                        familyName: contributor.family_name,
+                        givenName: contributor.given_name,
+                        id: contributor['@id']
+                    };
+                });
+                return source;
+            });
+            Ember.run(() => {
+                this.set('loading', false);
+                this.set('results', results);
+
+                this.togglePagination();
+            });
+        });
+    },
+
+    getQueryBody() {
+        let facetFilters = this.get('facetFilters');
+        let filters = [];
+        for (let k of Object.keys(facetFilters)) {
+            let filter = facetFilters[k];
+            if (filter) {
+                if (Ember.$.isArray(filter)) {
+                    filters = filters.concat(filter);
+                } else {
+                    filters.push(filter);
+                }
+            }
+        }
+
+        let query = {
+            'query_string' : {
+                'query': this.get('searchString') || '*'
+            }
+        };
+        if (filters.length) {
+            query = {
+                'bool': {
+                    'must': query,
+                    'filter': filters
+                }
+            };
+        }
+
+        let sort = [];
+        let sortBy = this.get('sortBy') || 'Relevance';
+        if ( sortBy === 'Upload date (oldest to newest)' ) {
+            sort.push({
+                "date_updated": {"order": "asc"}
+            });
+        } else if ( sortBy === 'Upload date (newest to oldest)' ) {
+            sort.push({
+                "date_updated": {"order": "desc"}
+            });
+        }
+
+        let queryBody = {
+            query,
+            from: (this.get('page') - 1) * this.get('size'),
+            sort
+        };
+
+        return this.set('queryBody', queryBody);
+    },
+
+    togglePagination() {
+        // prev
+        if (this.get('page') > 1) {
+            this.set('showPrev', true);
```

Comment:
You might consider having the showPrev and showNext variables be computeds. That way, you don't need to manually change them.

There are even some convenience methods to reduce boilerplate. 
http://emberjs.com/api/classes/Ember.computed.html#method_gt